ostree/ostree_transport.go

```diff
@@ -0,0 +1,181 @@
+type ostreeReference struct {
+	ref          string
+	repo         string
+	image        string
+}
+
+func NewReference(path string) (types.ImageReference, error) {
+	var repo = ""
+	var image = ""
+	s := strings.SplitN(path, "@", 2)
+	if len(s) == 1 {
+		image, repo = s[0], "/ostree/repo"
+	} else {
+		image, repo = s[0], s[1]
+	}
+	return ostreeReference{ref: path, repo: repo, image: image}, nil
+}
+
+func (ref ostreeReference) Transport() types.ImageTransport {
+	return Transport
+}
+
+// StringWithinTransport returns a string representation of the reference, which MUST be such that
+// reference.Transport().ParseReference(reference.StringWithinTransport()) returns an equivalent reference.
+// NOTE: The returned string is not promised to be equal to the original input to ParseReference;
+// e.g. default attribute values omitted by the user may be filled in in the return value, or vice versa.
+// WARNING: Do not use the return value in the UI to describe an image, it does not contain the Transport().Name() prefix.
+func (ref ostreeReference) StringWithinTransport() string {
+	return ref.ref
+}
+
+// DockerReference returns a Docker reference associated with this reference
+// (fully explicit, i.e. !reference.IsNameOnly, but reflecting user intent,
+// not e.g. after redirect or alias processing), or nil if unknown/not applicable.
+func (ref ostreeReference) DockerReference() reference.Named {
+	return nil
+}
+
+// PolicyConfigurationIdentity returns a string representation of the reference, suitable for policy lookup.
+// This MUST reflect user intent, not e.g. after processing of third-party redirects or aliases;
+// The value SHOULD be fully explicit about its semantics, with no hidden defaults, AND canonical
+// (i.e. various references with exactly the same semantics should return the same configuration identity)
+// It is fine for the return value to be equal to StringWithinTransport(), and it is desirable but
+// not required/guaranteed that it will be a valid input to Transport().ParseReference().
+// Returns "" if configuration identities for these references are not supported.
+func (ref ostreeReference) PolicyConfigurationIdentity() string {
+	return ref.ref
+}
+
+// PolicyConfigurationNamespaces returns a list of other policy configuration namespaces to search
+// for if explicit configuration for PolicyConfigurationIdentity() is not set.  The list will be processed
+// in order, terminating on first match, and an implicit "" is always checked at the end.
+// It is STRONGLY recommended for the first element, if any, to be a prefix of PolicyConfigurationIdentity(),
+// and each following element to be a prefix of the element preceding it.
+func (ref ostreeReference) PolicyConfigurationNamespaces() []string {
+	res := []string{}
+	path := ref.ref
+	for {
+		lastSlash := strings.LastIndex(path, "/")
+		if lastSlash == -1 || lastSlash == 0 {
+			break
+		}
+		path = path[:lastSlash]
+		res = append(res, path)
+	}
```

Comment:
This does not really make sense; for `whatever@/my/repo` this would return `whatever@/my`, `whatever@`.

Designing the hierarchy here so that it makes sense for uses is basically the major outstanding action item for `ostreeReference`/`ostreeTransport`.  (It would be possible to not have any parent namespaces, only a `PolicyConfigurationIdentity`.)